### ISSUE TITLE
Ctrl+C does not take effect while cd [tab] 

### ISSUE THREAD

**Issue by jianyunt** [PowerShell/PowerShell#3537]
PowerShell Providers have a property, called `Stopping`, which is defined in line 1756 [CmdletProvider\ProviderBase.cs](https://github.com/PowerShell/PowerShell/blob/master/src/System.Management.Automation/namespaces/ProviderBase.cs#L1756).  This is useful for a user to hit Ctrl+C to stop the current operation.

The problem is for a case like `cd [tab]`,  Stopping does not gets set. Therefore Ctrl+C takes no action.

It may not be a big deal for FileSystem, Env, or Cert providers. However if the datastore is large and requires network calls, and a user cd [tab] under a path,  the system becomes irresponsive. And Ctrl+C does not work. It can be frustrating to end users.

Steps to reproduce
------------------
1. launch PowerShell
2. dir  \\scratch2\scratch    
3. Ctrl+C
4. It will stop right away. Good.
5. cd \\scratch2\scratch\[tab]
6. you will notice that no response because there are lots of stuff in the share
7. Ctrl+C, does not do anything.

Expected behavior
-----------------
Should stop after Ctrl+C.

Actual behavior
---------------
Ctrl+C does not seem to do anything. I also noticed Stopping property in the CmdletProvider does not get set to true after pressing Ctrl+C.

Environment data
----------------

```powershell
Name                           Value
----                           -----
GitCommitId                    v6.0.0-alpha.18-9-g8d4db01a5d4b5dcd981ef216d3d7f24b484c7c60
PSRemotingProtocolVersion      2.3
WSManStackVersion              3.0
BuildVersion                   3.0.0.0
SerializationVersion           1.1.0.1
PSEdition                      Core
PSVersion                      6.0.0-alpha
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
CLRVersion

or 

Name                           Value
----                           -----
PSVersion                      5.1.14987.1000
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.14987.1000
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

**Comment by markosimulak** [comment]
Hi there. This is sounds pretty good for me.

**Comment by ThomasNieto** [comment]
@StevenBucher98 can this be closed, I can't reproduce the issue on 7.4.
